docs/reference/analysis/tokenizers.asciidoc

```diff
@@ -1,34 +1,135 @@
+_character offsets_ of the original word which the term represents (used foor
+highlighting search snippets).
 
-include::tokenizers/standard-tokenizer.asciidoc[]
+Elasticsearch has a number of built in tokenizers which can be used to build
+<<analysis-custom-analyzer,custom analyzers>>.
 
-include::tokenizers/edgengram-tokenizer.asciidoc[]
+[float]
+=== Word Oriented Tokenizers
 
-include::tokenizers/keyword-tokenizer.asciidoc[]
+The following tokenizers are usually used for tokenizing full text into
+individual words:
+
+<<analysis-standard-tokenizer,Standard Tokenizer>>::
+
+The `standard` tokenizer divides text into terms on word boundaries, as
+defined by the Unicode Text Segmentation algorithm. It removes most
+punctuation symbols. It is the best choice for most languages.
+
+<<analysis-letter-tokenizer,Letter Tokenizer>>::
+
+The `letter` tokenizer divides text into terms whenever it encounters a
+character which is not a letter.
+
+<<analysis-letter-tokenizer,Lowercase Tokenizer>>::
+
+The `lowercase` tokenizer, like the `letter` tokenizer,  divides text into
+terms whenever it encounters a character which is not a letter, but it also
+lowercases all terms.
+
+<<analysis-whitespace-tokenizer,Whitespace Tokenizer>>::
+
+The `whitespace` tokenizer divides text into terms whenever it encounters any
+whitespace character.
+
+<<analysis-uaxurlemail-tokenizer,UAX URL Email Tokenizer>>::
+
+The `uax_url_email` tokenizer is like the `standard` tokenizer except that it
+recognises URLs and email addresses as single tokens.
+
+<<analysis-classic-tokenizer,Classic Tokenizer>>::
+
+The `classic` tokenizer is a grammar based tokenizer for the English Language.
+
+<<analysis-thai-tokenizer,Thai Tokenizer>>::
+
+The `thai` tokenizer segments Thai text into words.
+
+[float]
+=== Partial Word Tokenizers
+
+These tokenizers break up text or words into small fragments, for partial word
+matching:
+
+<<analysis-ngram-tokenizer,N-Gram Tokenizer>>::
+
+The `ngram` tokenizer can break up text into words when it encounters any of
+a list of specified characters (e.g. whitespace or punctuation), then it returns
+n-grams of each word: a sliding window of continuous letters, e.g. `quick` ->
+`[qu, ui, ic, ck]`.
+
+<<analysis-edgengram-tokenizer,Edge N-Gram Tokenizer>>::
+
+The `edge_ngram` tokenizer can break up text into words when it encounters any of
+a list of specified characters (e.g. whitespace or punctuation), then it returns
+n-grams of each word which are anchored to the start of the word, e.g. `quick` ->
+`[q, qu, qui, quic, quick]`.
+
+
+[float]
+=== Structured Text Tokenizers
+
+The following tokenizers are usually used with structured text like
+identifiers, email addresses, zip codes, and paths, rather than with full
+text:
+
+<<analysis-keyword-tokenizer,Keyword Tokenizer>>::
+
+The `keyword` tokenizer is a ``noop'' tokenizer that accepts whatever text it
+is given and outputs the exact same text as a single term.  It can be combined
```

Comment:
I use term and token pretty much interchangeably.  Not sure if this is correct or not, or what the real distinction is